Added SpanishGame and AcesUpGame classes.

AntonSynytsiaOSU opened the pull request:
Generalized the Game class and added SpanishGame and AcesUpGame classes, which both inherit from the Game class.

lcsfrey commented on src/main/java/models/Game.java:
```diff
@@ -127,4 +95,9 @@ public boolean isCardMovable(int column) {
     }
+
+    // returns true if card at column can be removed
+    public boolean isCardRemovable(int column) {
```
Does this function need to be defined in the abstract class? I think declaring it and letting the subclasses define it would be sufficient. (Since all it's doing here is returning false.)

lcsfrey commented on src/main/java/models/Game.java:
```diff
@@ -127,4 +95,9 @@ public boolean isCardMovable(int column) {
+
+    // returns true if card at column can be removed
+    public boolean isCardRemovable(int column) {
+        return false;
```
Does this function need to be defined in the abstract class? I think declaring it and letting the subclasses define it would be sufficient. (Since all it's doing here is returning false.)

AntonSynytsiaOSU replied:
I didn't know it were possible to declare abstract methods in Java. This also applies to getScore. Will fix in next commit.

lcsfrey commented on src/main/java/models/SpanishGame.java:
```diff
@@ -0,0 +1,62 @@
+        // If card at column is not a Joker, and there is another card that is either of the same suit or is a Joker, we remove the card.
+        else if (isCardRemovable(column)) {
+            super.remove(column);
+            this.status = c1.toString() + " (Removed)";
```
Recall that when a Joker is used to remove a lower value card, the Joker is also removed. This function currently only removes the lower value card.

AntonSynytsiaOSU replied:
Fixed!

AntonSynytsiaOSU commented:
Alright, I updated the remove function. When a card at a column is clicked, it is removed if there is another card of the same suit and a higher value at a different column. If there is no higher card of the same suit, then the card is removed if and only if there is a Joker at a different column, in which case the Joker is also removed.

AntonSynytsiaOSU commented:
I finished with the game class and its tests as well. I also updated the customDeal function, to deal based on the list size rather than fixed size of 4.
